### ISSUE TITLE
Error when converting *.json to doc  

### ISSUE THREAD

**Issue by d5555** [explosion/spaCy#7635]
https://github.com/explosion/spaCy/blob/master/spacy/training/gold_io.pyx
line 124 in `def json_to_annotations(doc)`:  
It should be `sent_starts.append(-1) `
instead of `sent_starts.append(0) `

Otherwise when we create doc from json we get sentence_starts :   True, None, None, None ...
Then when I want to iterate sentences in `doc.sents` I get an error : [E030] Sentence boundaries unset.

**Comment by d5555** [comment]
Error appears when it is one sentence in the doc.     For instance `sentence_starts` : `True, None, None, None`
If there are 2 sentences in the doc then sentencizer works ok. `sentence_starts` : `True, None, None, True, None, None`

**Comment by polm** [comment]
Can you give a complete code example? I'm not entirely sure what you're trying to do or why it isn't working. This works for me:

```
import spacy
from spacy.training.gold_io import docs_to_json, json_to_annotations
from spacy.training import Example

nlp = spacy.load("en_core_web_sm")

doc = nlp("The rain in Spain falls mainly on the plain.")

json = docs_to_json([doc])

redoc = json_to_annotations(json)

for item in redoc:
    doc = nlp(item["text"])
    ex = Example.from_dict(doc, item)
    for sent in ex.predicted.sents:
        print(sent)
```

The logic in the code also looks fine - for each sentence in the document, the first token has a `True` sent start, and the others have `None`, which is valid.

**Comment by polm** [comment]
Ah, actually, you're right - this is an issue. Using `nlp.make_doc` in the above code triggers it. 

We'll work on fixing this, but for now you can work around it by setting `is_sent_start` to `False` if it's `None` before iterating the `sents` on your Doc.

**Comment by polm** [comment]
One question - can you provide more details about how you use the sentences iterator during training? We don't usually do that, so if we had an idea of how it could be useful we might be able to add a more thorough test case or something.

**Comment by d5555** [comment]
As a work around I just do checking  doc_.has_annotation("SENT_START") before iterating docs.  If  no, I set last token "False."

I needed to convert json to docs. If there was only one sentence in the doc I was getting an error. 
Something like this:

`from spacy.training.converters.json_to_docs import json_to_docs 
for doc in json_to_docs(json_file, model):
     for token in doc:
         print(token.text)`

Same happens in this example:
`doc = Doc(nlp.vocab, words=["hello", "world", "!"], sent_starts=[True, None, None])
for token in doc.sents:
	token.text`
